aopy/data/bmi3d.py

```diff
@@ -1117,6 +1117,91 @@ def get_ts_data_segment(preproc_dir, subject, te_id, date, start_time, end_time,
 
     return data, samplerate
 
+def get_spike_data_segment(preproc_dir, subject, te_id, date, port, start_time, end_time, bin_width=.01):
+    '''
+    Loads and extracts a segment of spiking data for a given subject and experiment, optionally binning the spike times.
+
+    Args:
+        preproc_dir (str): Path to the preprocessed data directory.
+        subject (str): Subject name.
+        te_id (str): Task entry number.
+        date (str): The date of the experiment.
+        port (int, optional): The port number corresponding to the spike data.
+        start_time (float): The start time [s] of the segment to extract.
+        end_time (float): The end time [s] of the segment to extract.
+        bin_width (float, optional): The width of the bins [s]. Default is 0.01 (10ms) seconds. If set to `None`, no binning is applied and spike times are returned.
+        
+    Returns:
+        tuple: A tuple containing:
+            - spike_segment (dict): A dictionary where keys are unit labels and values are arrays of spike times (or binned spike counts) for that unit.
+            - bins (numpy.ndarray or None): An array of bin edges if binning was applied, otherwise `None`.
+    
+    '''
+
+    # Load data
+    filename_mc = get_preprocessed_filename(subject, te_id, date, 'spike')
+    spike_data = load_hdf_group(os.path.join(preproc_dir, subject), filename_mc, f'drive{port}/spikes')
```

Comment:
i think you want to set cached=True here otherwise you're throwing away the spike data and reloading it each time for the same recording but different start and end times.

Reply:
with continuous data we load just the relevant section of data from hdf but i don't think that approach works for spike times